```
==> build (dialyzer)
riak_kv_wm_timeseries_query.erl:202: The call riak_kv_wm_timeseries_query:process_post_(RD::any(),Ctx::#ctx{api_version::'undefined' | integer(),table::<<>>,mod::atom() | tuple(),method::atom(),timeout::'undefined' | integer(),sql_type::'ddl' | 'describe' | 'explain' | 'insert' | 'select' | 'show_create_table' | 'show_tables' | 'undefined',compiled_query::'undefined' | #riak_sql_describe_v1{'DESCRIBE'::binary()} | #riak_sql_show_create_table_v1{'SHOW_CREATE_TABLE'::binary()} | #riak_sql_show_tables_v1{'SHOW_TABLES'::binary()} | #riak_sql_delete_query_v1{'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::'undefined' | [any()],helper_mod::atom()} | #riak_sql_insert_v1{'INSERT'::binary(),fields::'undefined' | [{'identifier',[binary()]}],values::'undefined' | [[{'binary',binary()} | {'boolean',boolean()} | {'float',float()} | {'integer',integer()}]],helper_mod::atom()} | #ddl_v2{table::'undefined' | binary(),fields::[#riak_field_v1{name::binary(),position::'undefined' | pos_integer(),type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar',optional::boolean()}],partition_key::'none' | 'undefined' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},local_key::'undefined' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},minimum_capability::'v1' | 'v2'} | #riak_select_v3{'SELECT'::'undefined' | #riak_sel_clause_v1{calc_type::'aggregate' | 'group_by' | 'rows',initial_state::[any()],col_return_types::['blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'varchar'],col_names::[binary()],clause::[fun((_,_) -> boolean() | binary() | [any()] | number())],finalisers::['skip' | fun()]},'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::[any()],'ORDER BY'::[{binary(),'asc' | 'desc','nulls_first' | 'nulls_last'}],'LIMIT'::[pos_integer()],helper_mod::atom(),partition_key::'none' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},is_executable::boolean(),type::'sql' | 'timeseries',group_by::[{'identifier' | integer(),binary()}],'OFFSET'::[non_neg_integer()],qbuf_id::'undefined' | binary()},with_props::'undefined' | [atom() | tuple()],result::'ok' | 'undefined' | {[binary()],[[boolean() | binary() | [any()] | number()]]}},SQL::'undefined' | #riak_sql_describe_v1{'DESCRIBE'::binary()} | #riak_sql_show_create_table_v1{'SHOW_CREATE_TABLE'::binary()} | #riak_sql_show_tables_v1{'SHOW_TABLES'::binary()} | #riak_sql_delete_query_v1{'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::'undefined' | [any()],helper_mod::atom()} | #riak_sql_insert_v1{'INSERT'::binary(),fields::'undefined' | [{'identifier',[binary()]}],values::'undefined' | [[{'binary',binary()} | {'boolean',boolean()} | {'float',float()} | {'integer',integer()}]],helper_mod::atom()} | #ddl_v2{table::'undefined' | binary(),fields::[#riak_field_v1{name::binary(),position::'undefined' | pos_integer(),type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar',optional::boolean()}],partition_key::'none' | 'undefined' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},local_key::'undefined' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},minimum_capability::'v1' | 'v2'} | #riak_select_v3{'SELECT'::'undefined' | #riak_sel_clause_v1{calc_type::'aggregate' | 'group_by' | 'rows',initial_state::[any()],col_return_types::['blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'varchar'],col_names::[binary()],clause::[fun((_,_) -> boolean() | binary() | [any()] | number())],finalisers::['skip' | fun()]},'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::[any()],'ORDER BY'::[{binary(),'asc' | 'desc','nulls_first' | 'nulls_last'}],'LIMIT'::[pos_integer()],helper_mod::atom(),partition_key::'none' | #key_v1{ast::[#param_v2{name::[binary()],ordering::'ascending' | 'descending' | 'undefined'} | #hash_fn_v1{mod::atom(),fn::atom(),args::[any()],type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar'}]},is_executable::boolean(),type::'sql' | 'timeseries',group_by::[{'identifier' | integer(),binary()}],'OFFSET'::[non_neg_integer()],qbuf_id::'undefined' | binary()},QueryType::'ddl' | 'describe' | 'explain' | 'insert' | 'select' | 'show_create_table' | 'show_tables' | 'undefined','undefined','undefined') will never return since it differs in the 6th argument from the success typing arguments: (#wm_reqdata{},#ctx{api_version::'undefined' | integer(),table::'undefined' | binary(),mod::atom() | tuple(),method::atom(),timeout::'undefined' | integer(),sql_type::'ddl' | 'describe' | 'explain' | 'insert' | 'select' | 'show_create_table' | 'show_tables' | 'undefined',compiled_query::'undefined' | #riak_sql_describe_v1{'DESCRIBE'::binary()} | #riak_sql_show_create_table_v1{'SHOW_CREATE_TABLE'::binary()} | #riak_sql_show_tables_v1{'SHOW_TABLES'::binary()} | #riak_sql_delete_query_v1{'FROM'::binary() | {_,_},'WHERE'::'undefined' | [any()],helper_mod::atom()} | #riak_sql_insert_v1{'INSERT'::binary(),fields::'undefined' | [any()],values::'undefined' | [any()],helper_mod::atom()} | #ddl_v2{table::'undefined' | binary(),fields::[any()],partition_key::'none' | 'undefined' | {_,_},local_key::'undefined' | {_,_},minimum_capability::'v1' | 'v2'} | #riak_select_v3{'SELECT'::'undefined' | {_,_,_,_,_,_,_},'FROM'::binary() | {_,_},'WHERE'::[any()],'ORDER BY'::[any()],'LIMIT'::[any()],helper_mod::atom(),partition_key::'none' | {_,_},is_executable::boolean(),type::'sql' | 'timeseries',group_by::[any()],'OFFSET'::[any()],qbuf_id::'undefined' | binary()},with_props::'undefined' | [atom() | tuple()],result::'ok' | 'undefined' | {[any()],[any()]}},#riak_sql_describe_v1{'DESCRIBE'::binary()} | #riak_sql_show_create_table_v1{'SHOW_CREATE_TABLE'::binary()} | #riak_sql_show_tables_v1{'SHOW_TABLES'::binary()} | #riak_sql_delete_query_v1{'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::'undefined' | [any()],helper_mod::atom()} | #riak_sql_insert_v1{'INSERT'::binary(),fields::[{'identifier',[any()]}],values::'undefined' | [[{_,_}]],helper_mod::atom()} | #riak_select_v3{'SELECT'::#riak_sel_clause_v1{calc_type::'aggregate' | 'group_by' | 'rows',initial_state::[any()],col_return_types::['blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'varchar'],col_names::[binary()],clause::[fun((_,_) -> any())],finalisers::['skip' | fun()]},'FROM'::binary() | {'list',[binary()]} | {'regex',[any()]},'WHERE'::[any()],'ORDER BY'::[{binary(),'asc' | 'desc','nulls_first' | 'nulls_last'}],'LIMIT'::[pos_integer()],helper_mod::atom(),partition_key::'none' | #key_v1{ast::[{_,_,_} | {_,_,_,_,_}]},is_executable::boolean(),type::'sql' | 'timeseries',group_by::[{'identifier' | integer(),binary()}],'OFFSET'::[non_neg_integer()],qbuf_id::'undefined' | binary()},'ddl' | 'describe' | 'explain' | 'insert' | 'select' | 'show_create_table' | 'show_tables' | 'undefined','undefined' | binary(),#ddl_v2{table::'undefined' | binary(),fields::[#riak_field_v1{name::binary(),position::'undefined' | pos_integer(),type::'blob' | 'boolean' | 'double' | 'sint64' | 'timestamp' | 'undefined' | 'varchar',optional::boolean()}],partition_key::'none' | 'undefined' | #key_v1{ast::[{_,_,_} | {_,_,_,_,_}]},local_key::'undefined' | #key_v1{ast::[{_,_,_} | {_,_,_,_,_}]},minimum_capability::'v1' | 'v2'})
make: *** [dialyzer-run] Error 1
```

--- a/src/riak_kv_wm_timeseries_query.erl
+++ b/src/riak_kv_wm_timeseries_query.erl
@@ -195,19 +195,18 @@ process_post(RD, #ctx{sql_type = ddl, compiled_query = SQL, with_props = WithPro
         {error, Reason} ->
             riak_kv_wm_ts_util:handle_error(Reason, RD, Ctx)
     end;
-process_post(RD, #ctx{sql_type = QueryType,
-                      compiled_query = SQL,
-                      table = <<>>} = Ctx) ->
+process_post(RD, #ctx{table = <<>>} = Ctx) ->
     % SHOW TABLES has table = <<>>
-    process_post_(RD, Ctx, SQL, QueryType, undefined, undefined);
-process_post(RD, #ctx{sql_type = QueryType,
-                      compiled_query = SQL,
-                      table = Table,
-                      mod = Mod} = Ctx) ->
+    process_post_(RD, Ctx, ?DDL{});
+process_post(RD, #ctx{mod = Mod} = Ctx) ->
     DDL = Mod:get_ddl(), %% might be faster to store this earlier on
-    process_post_(RD, Ctx, SQL, QueryType, Table, DDL).
+    process_post_(RD, Ctx, DDL).
 
-process_post_(RD, Ctx, SQL, QueryType, Table, DDL) ->
+-spec process_post_(#wm_reqdata{}, #ctx{},
+                    ?DDL{}) -> cb_rv_spec(boolean()).
+process_post_(RD, #ctx{sql_type = QueryType,
+                       compiled_query = SQL,
+                       table = Table} = Ctx, DDL) ->
     case riak_kv_ts_api:query(SQL, DDL) of
         {ok, Data} ->
             {ColumnNames, _ColumnTypes, Rows} = Data,

